### DISCUSSION TITLE
More details for Pro route limits

### DISCUSSION THREAD

**Discussion by SwissArmyMike** [vercel/community#2002] [Help]
### Problem Description

Am I understanding the route limits correctly? Currently deployments are limited to 1024 routes. According to the docs this includes dynamic and static routes. So this effectively means that any app or service that utilized multi-tenancy such as a service that let users create blog posts would quickly exceed the limitations if they go beyond 1024 posts. Is this correct?

**Comment by jk-tagbangers** [comment]
I'm on the Pro plan and have a same question.

My project uses Next.js with dynamic routing using headless CMS and 4 locales per page.
Over 400 routes have been generated and there are concerns about future route counts.

I tried routing more than 1024, but the deployment succeeded without any error, so I'm confused 🤔

```
info  - Collecting page data...
info  - Generating static pages (0/1566)
...
Build Completed in /vercel/output [1m]
Deploying outputs...
Generated build outputs:
- Static files: 53
- Prerenders: 3168
- Serverless Functions: 14
- Edge Functions: 1
```

I'd like to know in more detail in what cases routing restrictions occur in Vercel.
